lib/ContractInteract/Recovery.js

```diff
@@ -17,43 +16,91 @@ class Recovery {
       return Promise.reject(err);
     }
 
     this.auxiliaryWeb3 = auxiliaryWeb3;
-    this.delayedRecoveryAddress = delayedRecoveryAddress;
+    this.address = address;
 
-    this.contract = Contracts.getDelayedRecovery(this.auxiliaryWeb3, this.delayedRecoveryAddress);
+    this.contract = Contracts.getDelayedRecovery(this.auxiliaryWeb3, this.address);
 
     if (!this.contract) {
-      const err = new TypeError(`Could not load recovery contract for: ${this.delayedRecoveryAddress}`);
+      const err = new TypeError(`Could not load recovery contract for: ${this.address}`);
+      return Promise.reject(err);
+    }
+  }
+
+  /**
+   * Deploys Recovery master copy contract.
+   *
+   * @param {Web3} auxiliaryWeb3 Origin chain web3 object.
+   * @param {Object} txOptions Tx options.
+   *
+   * @returns {Promise<Recovery>} Promise containing the Recovery
+   *                              instance that has been deployed.
+   */
+  static async deploy(auxiliaryWeb3, txOptions) {
+    if (!txOptions) {
+      const err = new TypeError('Invalid transaction options.');
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError(`Invalid from address: ${txOptions.from}.`);
       return Promise.reject(err);
     }
+
+    const tx = Recovery.deployRawTx(auxiliaryWeb3);
+
+    return Utils.sendTransaction(tx, txOptions).then((txReceipt) => {
+      const address = txReceipt.contractAddress;
+      return new Recovery(auxiliaryWeb3, address);
+    });
+  }
+
+  /**
+   * Method which returns Tx object to deploy GnosisSafe master copy contract.
+   *
+   * @param {Web3} auxiliaryWeb3 Auxiliary chain web3 object.
+   *
+   * @returns {Object} Raw transaction object.
+   */
+  static deployRawTx(auxiliaryWeb3) {
+    if (!(auxiliaryWeb3 instanceof Web3)) {
+      throw new TypeError(`Mandatory Parameter 'auxiliaryWeb3' is missing or invalid: ${auxiliaryWeb3}`);
+    }
+    const abiBinProvider = new AbiBinProvider();
+    const bin = abiBinProvider.getBIN(ContractName);
+
+    const args = [];
+    const contract = Contracts.getDelayedRecovery(auxiliaryWeb3);
+
+    return contract.deploy({
+      data: bin,
+      arguments: args
+    });
   }
 
   /**
    * Initiates a recovery procedure.
    *
-   * @param prevOwner Owner that pointed to the owner to be replaced in the linked list.
-   * @param oldOwner Owner address to replace.
-   * @param newOwner New owner address.
-   * @param r `r` part of the signature.
-   * @param s `s` part of the signature.
-   * @param v `v` part of the signature.
-   * @param txOptions Transaction options object.
+   * @param {String} prevOwner Owner that pointed to the owner to be replaced in the linked list.
+   * @param {String} oldOwner Owner address to replace.
+   * @param {String} newOwner New owner address.
+   * @param {String} r `r` part of the signature.
```

Comment:
It's nice to know how this r,s,v is generated .what hash is signed to generate signature.